### DISCUSSION TITLE
Need to add an exception dictionary for `CLASS_NAME_INCORRECT`

### DISCUSSION THREAD

**Discussion by Cheshiriks** [cqfn/diKTat#1622] [Ideas]
It is not always correct to name classes only in pascalCase. The class name may contain an abbreviation, such as DTO, DAO, etc. which must be in upper case.

Perhaps you can add manually configured exceptions to the rule, which the user will add himself.

**Comment by petertrr** [comment]
I believe we always were intentionally converting these parts to PascalCase, i.e. HtmlLoader instead of HTMLLoader etc.
Also can be considered related to #1079

**Comment by Cheshiriks** [comment]
I looked at the projects on github and noticed that quite a few use uppercase abbreviations. It seemed to me wrong not to give a choice to the user here.

**Comment by 0x6675636b796f75676974687562** [comment]
> The class name may contain an abbreviation, such as DTO, DAO

The right naming would be `XyzDto` and `XyzDao`.

As both [EJ](https://www.goodreads.com/book/show/105099.Effective_Java_) and [our own specification](https://github.com/saveourtool/diktat/blob/master/info/guide/diktat-coding-convention.md#-111-identifiers-naming-conventions) suggest, `HttpUrlConnection` is a better name than `HTTPURLConnection`. See [here](https://medium.com/codex/effective-java-adhere-to-generally-accepted-naming-conventions-5e4abaf12571) and [here](http://dolszewski.com/java/java-class-naming-ultimate-guideline) for more details.

Kotlin's naming conventions are slightly different from those of Java, but they generally agree on how class names with acronyms at the beginning/in the middle should look like.
